### PR TITLE
Results Element filter

### PR DESCRIPTION
**Pull request by jthorton**
## Description
This PR adds a results element filter.

## Todos
Notable points that this PR has either accomplished or will accomplish.
  - [X] Make the element validation function a general validator that can be reused.

## Status
- [X] Ready to go

### CODE REVIEW

File: openff/qcsubmit/validators.py [hunk 1]
```diff
@@ -259,3 +259,27 @@ def check_valence_connectivity(molecule: qcel.models.Molecule) -> qcel.models.Mo
         )
 
     return molecule
+
+
+def check_allowed_elements(element: Union[str, int]) -> Union[str, int]:
```
Comment (SimonBoothroyd) [inline openff/qcsubmit/validators.py:264]: This is great - good catch that this would be duplicate code!

File: openff/qcsubmit/validators.py [hunk 2]
```diff
@@ -259,3 +259,27 @@ def check_valence_connectivity(molecule: qcel.models.Molecule) -> qcel.models.Mo
...
+    if isinstance(element, int):
+        return element
+    else:
+        try:
+            _ = Element.getBySymbol(element)
+            return element
+        except KeyError:
+            raise ValueError(
+                f"An element could not be determined from symbol {element}, please enter symbols only."
+            )
```
Comment (SimonBoothroyd) [inline openff/qcsubmit/validators.py:285]:
```suggestion
    try:
        _ = Element.getBySymbol(element)
        return element
    except KeyError:
        raise ValueError(
            f"An element could not be determined from symbol {element}, please enter symbols only."
        )
```

We can get away without the `else` statement here as the `if` statement returns when it is triggered. This lets us un-nest the larger code block which usually helps with readability 🙂

**Review by SimonBoothroyd — APPROVED** [review]
This is a great addition! I noticed just a few little nit-picky lint changes but otherwise this LGTM!

File: openff/qcsubmit/validators.py (hunk 2 above)
Comment (jthorton) [reply openff/qcsubmit/validators.py:285]: Ah, good spot thanks!

**Pull request merged**